cpp/CMakeLists.txt

```diff
@@ -268,3 +268,22 @@ rapids_export(
   NAMESPACE cuspatial::
   DOCUMENTATION doc_string
 )
+
+
+# ##################################################################################################
+# * build documentation ----------------------------------------------------------------------------
+
```

Comment:
I think this also is roughly equal to the `doxygen_add_docs` function that `find_package(Doxygen)` provides so it might be worth using that: https://cmake.org/cmake/help/latest/module/FindDoxygen.html#functions

Reply:
Hmmm

> The function constructs a Doxyfile and defines a custom target that runs Doxygen on that generated file.

But we already have a Doxyfile, I don't want to create a new one.

Reply:
I took your first suggestion. If there's a command that can replace ours without regenerating Doxyfile I'll use it.